<?xml version="1.0" ?>
<launch>
    <!-- <arg name="robot_type" default="$(env ROBOT_TYPE)" doc="Robot type: [elspider_air]"/> -->
    <arg name="robot_type" default="elspider_air" doc="Robot type: [elspider_air]"/>

    <arg name="cheater" default="false"/>

    <!-- make the files into global parameters -->
    <param name="urdfFile" value="/tmp/legged_control/$(arg robot_type).urdf"/>
    <param name="taskFile" value="$(find legged_controllers)/config/$(arg robot_type)/task.info"/>
    <param name="referenceFile" value="$(find legged_controllers)/config/$(arg robot_type)/reference.info"/>
    <param name="gaitCommandFile" value="$(find legged_controllers)/config/$(arg robot_type)/gait.info"/>

    <rosparam file="$(find legged_controllers)/config/controllers.yaml" command="load"/>

    <!-- FIXME: Cheater -->
    <node if="$(arg cheater)" name="controller_loader" pkg="controller_manager" type="controller_manager" output="screen" args="load
            controllers/joint_state_controller
    "/>

    <node unless="$(arg cheater)" name="controller_loader" pkg="controller_manager" type="controller_manager" output="screen" args="load
            controllers/joint_state_controller
            controllers/hexapod_controller
    "/>

    <!-- BUG: do not support hexapod -->
    <node pkg="legged_reference" type="hexapod_gait_command_publisher" name="hexapod_gait_command_publisher" output="screen"/>

    <node pkg="legged_controllers" type="legged_target_trajectories_publisher" name="legged_robot_target" output="screen"/>
</launch>
